### ISSUE TITLE
failed to send notifications - [::1]:53: socket: permission denied

### ISSUE THREAD

**Issue by markcs** [containrrr/shoutrrr#338]
I'm sure this is a local issue, but I have not seen this with any other application.

When trying to send a notification to any service, I am getting the following error (ie for Pushover):

`failed to send notifications to pushover devices: Post "https://api.pushover.net/1/messages.json": dial tcp: lookup api.pushover.net on [::1]:53: dial udp [::1]:53: socket: permission denied`

and same for Discord
`failed to send discord notification: Post "https://discord.com/api/webhooks/<redacted>": dial tcp: lookup discord.com on [::1]:53: dial udp [::1]:53: socket: permission denied`

I've tried on two machines, both are running Ubuntu and same error.  Any tips appreciated

**Comment by piksel** [comment]
Well, the error itself is that it's trying to make DNS lookup of the domain name `api.pushover.net` through the default DNS server (localhost, which is being resolved as :::1, which is the IPv6 equivalent of IPv4's 127.0.0.1). For some reason it's denied to open a socket to the DNS server though.
To investigate any further, we would probably need to know more about the environment in which you are running shoutrrr.

**Comment by markcs** [comment]
Thanks for the reply.  What info can I provide ?

My symptoms seem very similar to [this](https://askubuntu.com/questions/1235805/ipv6-dns-resolve-ubuntu-20-04).  My system fails on the same address in that issue.  The issue linked was solved in the GUI, but I'm not sure how to do this from the server.  I think I use nmcli, but not sure of the format.

```
$ resolvectl status eno1
Link 2 (eno1)
    Current Scopes: DNS
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 192.169.1.110
       DNS Servers: 192.169.1.110 1.1.1.1
```

192.168.1.110 is my local Adguard.

**Comment by markcs** [comment]
I built shoutrrr from source and it works.

This appears to be a snap issue, which I need to investigate.

**Comment by piksel** [comment]
Yeah, that seems likely interesting. Maybe we shouldn't close this. @simskij, perhaps you have some insights? I have not even tried the snap version of shoutrrr.

**Comment by simskij** [comment]
Yeah, you're probably right. Being a strictly confined snap, it is probably missing some interface. I'll see if I can find some easy way of resolving this.
